### PR TITLE
Enable build and test

### PR DESCRIPTION
**Pull request by marun**
Dependencies vendored with [glide](https://glide.readthedocs.io/en/latest/).

### CODE REVIEW

**Comment by marun** [comment]
@sttts FYI

File: build/root/Makefile.generated_files [hunk 1]
```diff
@@ -32,6 +32,9 @@ endif
 # It's necessary to set this because some environments don't link sh -> bash.
 SHELL := /bin/bash
 
+# TODO(marun) target repo-infra's version of this file
```
Comment (sttts) [inline build/root/Makefile.generated_files]: I hope simplifying the build system is on the roadmap. It hurts to see the complexity moved over.

**Comment by irfanurrehman** [comment]
@marun, I guess the bazel build failed because of a missing tag. I did push a temporary tag today morning on the commits already in master. You might need to add that to your branch too.. !

File: build/root/Makefile.generated_files (hunk 1 above)
Comment (marun) [reply build/root/Makefile.generated_files]: That's definitely on the table.  Maintaining k/k's build system in the context of k/f won't be sustainable.  Deprecating make in favor of bazel is one possibility.

**Comment by shashidharatd** [comment]
This looks good to me except the vendoring part which vendors some unwanted files. but we can take care of this in the near future. what is important today is to enable development on federation as soon as possible.
/lgtm

**Comment by irfanurrehman** [comment]
Seems ok to me too, with the current state of things.
/lgtm
/approve

**Comment by irfanurrehman** [comment]
Merging this manually. 
Verified that unit and integration tests pass on this locally.
`pull-federation-bazel-build` will be turned off for the time being as we are using the release tars from `make quick-release` path in CI jobs.
`pull-federation-e2e-gce` will not pass on this PR as there are additional changes which are needed and will be submitted as a separate PR.

**Pull request merged**